services/scribe/backfill/contract.go

```diff
@@ -339,30 +338,28 @@ func (c *ContractBackfiller) getLogs(parentCtx context.Context, startHeight, end
 			select {
 			case <-ctx.Done():
 				LogEvent(ErrorLevel, "Context canceled while getting log", LogData{"cid": c.chainConfig.ChainID, "sh": startHeight, "eh": endHeight, "e": ctx.Err()})
-				doneChan <- false
+				errChan <- ctx.Err().Error()
 				return
-			case logInfos := <-rangeFilter.GetLogChan():
-				for _, log := range logInfos.logs {
+			case logChunks := <-logFetcher.logs:
+				for _, log := range logChunks {
 					logsChan <- log
 				}
-			default:
-				if rangeFilter.Done() {
-					finLogs, err := rangeFilter.Drain(ctx)
-					LogEvent(ErrorLevel, "range filter drained", LogData{"cid": c.chainConfig.ChainID, "ca": c.contractConfig.Address, "len": len(finLogs), "err": err})
-
-					for _, log := range finLogs {
-						LogEvent(ErrorLevel, "Log filter done, adding log to chan", LogData{"cid": c.chainConfig.ChainID, "bn": log.BlockNumber, "tx": log.TxHash.Hex(), "la": log.Address.String(), "ca": c.contractConfig.Address})
-
-						logsChan <- log
-					}
-					doneChan <- true
+			case <-logFetcher.doneChan:
+				remLogChunks, err := logFetcher.Drain(ctx)
+				if err != nil {
+					LogEvent(ErrorLevel, "Error draining logs", LogData{"cid": c.chainConfig.ChainID, "sh": startHeight, "eh": endHeight, "e": err.Error()})
 					return
 				}
+				for _, log := range remLogChunks {
+					logsChan <- log
+				}
+				close(logsChan)
```

Comment:
Closing the logsChan, once the channel is empty, the consuming goroutine at the top of this file exits its loop.